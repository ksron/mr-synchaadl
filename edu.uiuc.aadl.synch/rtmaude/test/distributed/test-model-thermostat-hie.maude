set show timing off .

load ../../interpreter-distributed.maude
load ../syntax/module-thermostat-hie.maude 

---------------------------------------------
mod TEST-THERMOSTAT-SYSTEM is
  including THERMOSTAT-SYSTEM-SYNTAX .
  including MODEL-TRANSITION-SYSTEM .

  var T : Time .

  op initState : -> Object .
  eq initState = initialize(model) .

  op cinitState : -> Object .
  eq cinitState = initialize(collapse(model)) .
endm

--- set print attribute on .

rew  [100] {initState | 0 | infty} .

rew  [100] {cinitState | 0 | infty} .

--- the number of reachable states up to 5 synchronous step
search {initState | 0 | 25} =>* {none | T | 25} .

search {cinitState | 0 | 25} =>* {none | T | 25} .
