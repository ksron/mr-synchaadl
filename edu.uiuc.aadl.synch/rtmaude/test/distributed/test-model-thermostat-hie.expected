==========================================
rewrite [100] in TEST-THERMOSTAT-SYSTEM : {initState | 0 | infty} .
rewrites: 8582
result GlobalSystem: {< top : System | properties : none,features : none,
    subcomponents : (< csys : System | properties : none,features : (< on :
    AsyncDataOutPort | properties : none,content : * # [(false).Bool],
    destination : (environment .. on) > < off : AsyncDataOutPort | properties :
    none,content : * # [(false).Bool],destination : (environment .. off) > <
    temp : AsyncDataInPort | properties : none,content : [0.0] # [(
    false).Bool],cache : [0.0] # [(false).Bool],destination : (cproc .. temp)
    >),subcomponents : < cproc : Process | properties : none,features : (< on :
    AsyncDataOutPort | properties : none,content : * # [(false).Bool],
    destination : on > < off : AsyncDataOutPort | properties : none,content : *
    # [(false).Bool],destination : off > < temp : AsyncDataInPort | properties
    : none,content : [0.0] # [(false).Bool],cache : [0.0] # [(false).Bool],
    destination : (cthread .. temp) >),subcomponents : < cthread : AsyncThread
    | properties : ((TimingProperties::Period => {{5}}) ; (
    HybridSynchAADL::SamplingTime => {{0.0 .. 2.0}}) ;
    HybridSynchAADL::ResponseTime => {{3.0 .. 4.0}}),features : (< on :
    AsyncDataOutPort | properties : none,content : * # [(false).Bool],
    destination : on > < off : AsyncDataOutPort | properties : none,content : *
    # [(false).Bool],destination : off > < temp : AsyncDataInPort | properties
    : none,content : [0.0] # [(false).Bool],cache : [2.1050000000000001e+1] # [
    (true).Bool],destination : empty >),subcomponents : (< lower : Data |
    properties : DataModel::InitialValue => {{10}},features : none,
    subcomponents : none,connections : empty,value : [1.0e+1] # [(true).Bool] >
    < upper : Data | properties : DataModel::InitialValue => {{30}},features :
    none,subcomponents : none,connections : empty,value : [3.0e+1] # [(
    true).Bool] > < curr : Data | properties : DataModel::InitialValue => {{
    15}},features : none,subcomponents : none,connections : empty,value : [
    2.0064843750000001e+1] # [(true).Bool] >),connections : empty,varGen : <
    "top.cthread" >,currState : init,completeStates : init,variables : x :
    Real,transitions : init -[on dispatch]-> init {(v{x} := ((c[curr] + f[
    temp]) / [[2]])) ; if(v[x] < c[lower])(on !) (elsif(v[x] > c[upper])off !)
    end if ; c{curr} := v[x]},loopBound : 0,transBound : 2,period : 5,perTimer
    : 13/8,msgTimer : infty,execTime : 5/2,execTimer : infty > > > <
    environment : AsyncEnv | properties : ((TimingProperties::Period => {{5}})
    ; HybridSynchAADL::MaxClockDeviation => {{1.0000000000000001e-1}}),features
    : (< on : AsyncEnvInPort | properties : none,content : * # [(false).Bool],
    envCache : null(Unit),target : csys,destination : empty > < off :
    AsyncEnvInPort | properties : none,content : * # [(false).Bool],envCache :
    null(Unit),target : csys,destination : empty > < out : AsyncEnvOutPort |
    properties : none,content : [0.0] # [(false).Bool],envCache : null(Real),
    target : csys,destination : (csys .. temp) >),subcomponents : < x : Data |
    properties : DataModel::InitialValue => {{16}},features : none,
    subcomponents : none,connections : empty,value : [2.1675000000000001e+1] #
    [(true).Bool] >,connections : x ==> out,varGen : < "top.environment" >,
    currMode : init,jumps : ((init -[on]-> state) ; (init -[off]-> init) ; (
    state -[on]-> state) ; state -[off]-> init),flows : ((init [x(t)= (c[x] + (
    v[t] * [[2.0000000000000001e-1]]))]) ; state [x(t)= (c[x] - (v[t] * [[
    1.0000000000000001e-1]]))]),sampling : csys :(0,2),response : csys :(3,4),
    period : 5,perTimer : 13/8,msgTimer : infty,samplingTimes : csys : (1/4,
    3/4, 5/4, 7/4),responseTimes : csys : (25/8, 27/8, 29/8, 31/8),envState :
    idle,envTimer : 0 >) > | 227/8 | infty}
==========================================
rewrite [100] in TEST-THERMOSTAT-SYSTEM : {cinitState | 0 | infty} .
rewrites: 7404
result GlobalSystem: {< top : System | properties : none,features : none,
    subcomponents : (< environment : AsyncEnv | properties : ((
    TimingProperties::Period => {{5}}) ; HybridSynchAADL::MaxClockDeviation =>
    {{1.0000000000000001e-1}}),features : (< on : AsyncEnvInPort | properties :
    none,content : * # [(false).Bool],envCache : null(Unit),target : csys,
    destination : empty > < off : AsyncEnvInPort | properties : none,content :
    * # [(false).Bool],envCache : null(Unit),target : csys,destination : empty
    > < out : AsyncEnvOutPort | properties : none,content : [0.0] # [(
    false).Bool],envCache : null(Real),target : csys,destination : ((csys .
    cproc . cthread) .. temp) >),subcomponents : < x : Data | properties :
    DataModel::InitialValue => {{16}},features : none,subcomponents : none,
    connections : empty,value : [2.1675000000000001e+1] # [(true).Bool] >,
    connections : x ==> out,varGen : < "top.environment" >,currMode : init,
    jumps : ((init -[on]-> state) ; (init -[off]-> init) ; (state -[on]->
    state) ; state -[off]-> init),flows : ((init [x(t)= (c[x] + (v[t] * [[
    2.0000000000000001e-1]]))]) ; state [x(t)= (c[x] - (v[t] * [[
    1.0000000000000001e-1]]))]),sampling : csys :(0,2),response : csys :(3,4),
    period : 5,perTimer : 13/8,msgTimer : infty,samplingTimes : csys : (1/4,
    3/4, 5/4, 7/4),responseTimes : csys : (25/8, 27/8, 29/8, 31/8),envState :
    idle,envTimer : 0 > < csys . cproc . cthread : AsyncThread | properties : (
    (TimingProperties::Period => {{5}}) ; (HybridSynchAADL::SamplingTime => {{
    0.0 .. 2.0}}) ; HybridSynchAADL::ResponseTime => {{3.0 .. 4.0}}),features :
    (< on : AsyncDataOutPort | properties : none,content : * # [(false).Bool],
    destination : (environment .. on) > < off : AsyncDataOutPort | properties :
    none,content : * # [(false).Bool],destination : (environment .. off) > <
    temp : AsyncDataInPort | properties : none,content : [0.0] # [(
    false).Bool],cache : [2.1050000000000001e+1] # [(true).Bool],destination :
    empty >),subcomponents : (< lower : Data | properties :
    DataModel::InitialValue => {{10}},features : none,subcomponents : none,
    connections : empty,value : [1.0e+1] # [(true).Bool] > < upper : Data |
    properties : DataModel::InitialValue => {{30}},features : none,
    subcomponents : none,connections : empty,value : [3.0e+1] # [(true).Bool] >
    < curr : Data | properties : DataModel::InitialValue => {{15}},features :
    none,subcomponents : none,connections : empty,value : [
    2.0064843750000001e+1] # [(true).Bool] >),connections : empty,varGen : <
    "top.cthread" >,currState : init,completeStates : init,variables : x :
    Real,transitions : init -[on dispatch]-> init {(v{x} := ((c[curr] + f[
    temp]) / [[2]])) ; if(v[x] < c[lower])(on !) (elsif(v[x] > c[upper])off !)
    end if ; c{curr} := v[x]},loopBound : 0,transBound : 2,period : 5,perTimer
    : 13/8,msgTimer : infty,execTime : 5/2,execTimer : infty >) > | 227/8 |
    infty}
==========================================
search in TEST-THERMOSTAT-SYSTEM : {initState | 0 | 25} =>* {none | T | 25} .

No solution.
states: 57817  rewrites: 4025605
==========================================
search in TEST-THERMOSTAT-SYSTEM : {cinitState | 0 | 25} =>* {none | T | 25} .

No solution.
states: 57817  rewrites: 3458757
Bye.
